### DISCUSSION TITLE
How can I open the dashboard on new tab?

### DISCUSSION THREAD

**Discussion by linsui** [folke/snacks.nvim#142] [Q&A]
When LazyVim uses dashboard.nvim I was able to do this with

```lua
 vim.api.nvim_create_autocmd("TabNewEntered", {
   callback = function()
     require('dashboard'):instance()
   end,
 })
 ```

but this doesn't work because it open the dashboard in a float window.

```lua
 vim.api.nvim_create_autocmd("TabNewEntered", {
   callback = function()
     Snacks.dashboard()
   end,
 })
 ```

How can I do the same thing? Thanks!

**Comment by folke** [comment]
Feel free to open a feature request (for the `win` module)

**Reply by linsui** [reply]
https://github.com/folke/snacks.nvim/issues/145 Thanks!

**Comment by dpetka2001** [comment]
>  vim.api.nvim_create_autocmd("TabNewEntered", {
   callback = function()
     Snacks.dashboard()
   end,
 })

This works perfectly fine for me. Screencast following

[Screencast_2024-11-22-19-42-30.webm](https://github.com/user-attachments/assets/5974a378-8d78-4f1e-b09a-deb8250fbb00)

**Reply by folke** [reply]
Right, I misread the question. I thought op wanted to open the dashboard in a new tab :)

**Reply by linsui** [reply]
Yes, I want to open the dashboard in a new tab.

**Reply by dpetka2001** [reply]
In the screencast I posted in my previous comment, I am opening the dashboard in a new tab when it's opened. I press `<leader><tab><tab>` to create a new tab and it automatically opens the dashboard. Is that not what you want to achieve or am I misunderstanding something?

**Reply by IamSmeagol** [reply]
Right now it opens the dashboard and hides all the tabs, but it doesn't replace the new tab. I think they want the dashboard to be inside the tab rather than on top -- as op said "float window"

**Reply by dpetka2001** [reply]
Right, when you hit `<esc>` it does show the buffer underneath. In that sense it is indeed a "float window".

But I fail to understand the use case of such behavior. The dashboard is still opened fullscreen when you open a new tab and then presumably you want to execute an action from the dashboard to do something.

**Reply by linsui** [reply]
It's not fullscreen for me. Not sure which config changes that.

**Reply by dpetka2001** [reply]
In the screencast that I posted it opens fullscreen for me with default LazyVim installation. Something is going on in your personal configuration would be my guess.

**Reply by linsui** [reply]
I guess so. I'll try to find it out, thanks!

**Reply by dpetka2001** [reply]
If you have customized the snacks configuration, try commenting out everything and see what happens with the autocmd that you posted. Also make sure that `snacks` is up to date, as what you're describing used to be an old behavior because of the dashboard styles configuration but it has been changed to open fullscreen since then.

**Reply by folke** [reply]
@linsui you need to update snacks

**Reply by linsui** [reply]
It works now, thanks!

**Comment by linsui** [comment]
With `Snacks.dashboard({ buf = 0, win = 0 })` it opens the dashboard correctly. But somehow it breaks the https://github.com/tiagovla/scope.nvim. When I open a file from the picker on the dashboard the file is not opened in the current tab but in the old tab. If I close the dashboard, then I can open the file in the current tab.

**Reply by folke** [reply]
That's the whole reason why I implemented it so it opens in a floating window instead.
If you want to use the `buf=0` option, you're on your own.
That option is internally only used during startup.

**Reply by linsui** [reply]
Hmmm, so maybe I should use the dashboard.nvim extra instead? I have no idea how they work at all... Or I can somehow use the buf in the new tab so that opened files are also in the new tab?
